File: estDelivery.py
from datetime import datetime, timedelta

def generate_delivery_interval(min_days=1, max_days=14, skip_weekends=True):
    start = datetime.now()
    
    # Calculate minimum delivery date (starting tomorrow)
    min_date = start + timedelta(days=min_days)
    
    # Calculate maximum delivery date
    max_date = start
    if skip_weekends:
        days_added = 0
        while days_added < max_days:
            max_date += timedelta(days=1)
            if max_date.isoweekday() < 6:  # Monday=1, Sunday=7
                days_added += 1
    else:
        max_date += timedelta(days=max_days)
    
    # Format dates - show day and date
    min_formatted = min_date.strftime('%a, %-d')  # Thu, 27
    max_formatted = max_date.strftime('%a, %b %-d')  # Thu, Sep 9
    
    return f"Est. delivery {min_formatted} - {max_formatted}"

# Test the function
print(generate_delivery_interval())

File: estDelivery.php
<?php
// Add this RIGHT AFTER your database connection code in item-detail.php

function generateDeliveryInterval($minDays = 1, $maxDays = 14, $skipWeekends = true) {
    $start = new DateTime();
    
    // Calculate minimum delivery date (starting tomorrow)
    $minDate = clone $start;
    $minDate->add(new DateInterval('P1D')); // Start from tomorrow
    
    // Calculate maximum delivery date 
    $maxDate = clone $start;
    if ($skipWeekends) {
        // Add business days only
        $daysAdded = 0;
        while ($daysAdded < $maxDays) {
            $maxDate->add(new DateInterval('P1D'));
            
            if ($maxDate->format('N') < 6) { // Monday = 1, Sunday = 7
            
                $daysAdded++;
            }
        }
    } else {
        $maxDate->add(new DateInterval('P' . $maxDays . 'D'));
    }
    
    // Format dates - show day and date
    $minFormatted = $minDate->format('D, j'); // Thu, 27
    
    $maxFormatted = $maxDate->format('D, M j'); // Thu, Sep 9
    
    return "Est. delivery " . $minFormatted . " - " . $maxFormatted;
}

// Test the function (you can remove this after testing)
 echo generateDeliveryInterval(); // Will show: "Est. delivery Tue, 27 - Thu, Sep 12"
?>
